### PR TITLE
added github actions workflows

### PR DESCRIPTION
**Pull request by AssassinTee**
Added starting point for github actions. This PR contains 3 files:
A workflow for
- Unix (Ubuntu)
- Windows (64 and 32)
- MacOS

I updated the README, too, so a Badge is shown for each workflow.
Currently the workflow just runs on push on master.
@heinrich5991 I guess you have to review this

### CODE REVIEW

**Comment by Dune-jr** [comment]
OSX and Windows should be built both with bam and cmake.
The name of the build targets should be more consistent

**Comment by AssassinTee** [comment]
suggestions for the build target name?

**Comment by jxsl13** [comment]
Unix -> Linux

**Comment by AssassinTee** [comment]
Note: I will squash the commits to one after i finished it. I have to push it to github for testing

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,7 +9,7 @@ jobs:
           command: |
             apt-get update -y
             apt-get install cmake libfreetype6-dev libsdl2-dev -y
-            git submodule update --init
+            git submodule update --init
```
Comment (Ryozuki) [inline .circleci/config.yml]: leading whitespace

Comment (AssassinTee) [reply .circleci/config.yml]: there went something very wrong, this file shouldn't be changed in the first place...

**Comment by oy** [comment]
Fixed by #2369

**Pull request closed without merging**